darwin64 install issues

I have downloaded from github the pre-compiled darwin64 files and try to follow your install directions for macos as follows but I get errors on first line.

xxxx:browserpass-darwin64-3.0.4 XXXX$ ls -la
total 8360
drwxr-xr-x@  8 xxxx  staff      256 Apr  9 12:26 .
drwx------+ 30 xxxx  staff      960 Apr  9 12:21 ..
-rw-r--r--@  1 xxxx  staff     8196 Apr  9 12:26 .DS_Store
-rw-r--r--@  1 xxxx  staff      757 Mar 12 17:39 LICENSE
-rw-r--r--@  1 xxxx  staff    20669 Apr  9 11:32 Makefile
-rw-r--r--@  1 xxxx  staff    13398 Apr  8 18:24 README.md
drwxr-xr-x@  5 xxxx  staff      160 Apr  4 04:43 browser-files
-rwxr-xr-x@  1 xxxx  staff  4219128 Apr  9 11:58 browserpass-darwin64

xxxxx:browserpass-darwin64-3.0.4 XXXX$ make BIN=browserpass-darwin64 configure
sed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/chromium-host.json
sed: 1: "browser-files/chromium- ...": undefined label 'rowser-files/chromium-host.json'
make: *** [configure] Error 1

xxxx:browserpass-darwin64-3.0.4 XXXX$ sudo make BIN=browserpass-darwin64 install
install -Dm755 -t "/usr/bin/" browserpass-darwin64
install: illegal option -- D
usage: install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 file2
       install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 ... fileN directory
       install -d [-v] [-g group] [-m mode] [-o owner] directory ...
make: *** [install] Error 64

With both these errors running the commands as you request, I cannot proceed, if an error on my end, please do tell and I will correct, otherwise, please advise on next steps to resolve and get browserpass installed correctly.

Nope, ran, brew install coreutils, it said there was no changes and completed. I then delete the unpack folder and extracted from archive fresh, then ran this command in the folder, make BIN=browserpass-darwin64 configure, received the same error as before and noted below. See the bold below, is that correct or a typo in the code somewhere?

xxxx:browserpass-darwin64-3.0.4 xxxx$ make BIN=browserpass-darwin64 configure
sed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/chromium-host.json
sed: 1: "browser-files/chromium- ...": undefined label '**rowser-files/chromium-host.json**'
make: *** [configure] Error 1

did that, error below, no gsed

xxxx:browserpass-darwin64-3.0.4 xxxxx$ make BIN=browserpass-darwin64 configure
gsed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/chromium-host.json
/bin/sh: gsed: command not found
make: *** [configure] Error 127

nothing in coreutils for sed or gsed, installed gnu-sed, sed and gsed there, but still errors on the configure and install commands.

Listing for gnu-sed

lmba:browserpass-darwin64-3.0.4 larrybeck$ brew ls -v gnu-sed
find /usr/local/Cellar/gnu-sed/4.7 -not -type d -print
/usr/local/Cellar/gnu-sed/4.7/INSTALL_RECEIPT.json
/usr/local/Cellar/gnu-sed/4.7/bin/gsed
/usr/local/Cellar/gnu-sed/4.7/.brew/gnu-sed.rb
/usr/local/Cellar/gnu-sed/4.7/ChangeLog
/usr/local/Cellar/gnu-sed/4.7/AUTHORS
/usr/local/Cellar/gnu-sed/4.7/libexec/man
/usr/local/Cellar/gnu-sed/4.7/libexec/gnubin/sed
/usr/local/Cellar/gnu-sed/4.7/libexec/gnuman/man1/sed.1
/usr/local/Cellar/gnu-sed/4.7/README
/usr/local/Cellar/gnu-sed/4.7/COPYING
/usr/local/Cellar/gnu-sed/4.7/NEWS
/usr/local/Cellar/gnu-sed/4.7/share/man/man1/gsed.1
/usr/local/Cellar/gnu-sed/4.7/share/info/sed.info

When using gsed and ginstall in makefile

configure command:
lmba:browserpass-darwin64-3.0.4 larrybeck$ make BIN=browserpass-darwin64 configure
gsed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/chromium-host.json
gsed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/firefox-host.json

install command:
lmba:browserpass-darwin64-3.0.4 larrybeck$ sudo make BIN=browserpass-darwin64 install
ginstall -Dm755 -t "/usr/bin/" browserpass-darwin64
ginstall: cannot create regular file '/usr/bin/browserpass-darwin64': Operation not permitted
make: *** [install] Error 1

when changing makefile back to sed and install, the following error on configure command:
lmba:browserpass-darwin64-3.0.4 larrybeck$ make BIN=browserpass-darwin64 configure
sed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/chromium-host.json
sed: 1: "browser-files/chromium- ...": undefined label 'rowser-files/chromium-host.json'
make: *** [configure] Error 1

Copied the new makefile, tried running both configure and install with sudo and no sudo, below are the results, neither work correctly for install, one appears to work and one not for configure commands.

Non-sudo:

lmba:browserpass-darwin64-3.0.4 larrybeck$ make BIN=browserpass-darwin64 configure
/usr/local/bin/gsed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/chromium-host.json
/usr/local/bin/gsed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/firefox-host.json

lmba:browserpass-darwin64-3.0.4 larrybeck$ make BIN=browserpass-darwin64 install
/usr/local/bin/ginstall -Dm755 -t "/usr/bin/" browserpass-darwin64
ginstall: cannot create regular file '/usr/bin/browserpass-darwin64': Operation not permitted
make: *** [install] Error 1

sudo:

lmba:browserpass-darwin64-3.0.4 larrybeck$ sudo make BIN=browserpass-darwin64 configure
/usr/local/bin/gsed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/chromium-host.json
/usr/local/bin/gsed -i "s|%%replace%%|/usr/bin/browserpass-darwin64|" browser-files/firefox-host.json

lmba:browserpass-darwin64-3.0.4 larrybeck$ sudo make BIN=browserpass-darwin64 install
/usr/local/bin/ginstall -Dm755 -t "/usr/bin/" browserpass-darwin64
ginstall: cannot create regular file '/usr/bin/browserpass-darwin64': Operation not permitted
make: *** [install] Error 1

when i run this, sudo make hosts-chromium, i get the following:

lmba:browserpass-darwin64-3.0.4 larrybeck$ sudo make hosts-chromium
make: getcwd: No such file or directory
make: *** No rule to make target `hosts-chromium'.  Stop.

--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,10 @@ GO_LDFLAGS := "-extldflags ${LDFLAGS}"
 APP_ID := com.github.browserpass.native
 OS := $(shell uname -s)
 
+# GNU tools
+SED := $(shell which gsed 2>/dev/null || which sed 2>/dev/null)
+INSTALL := $(shell which ginstall 2>/dev/null || which install 2>/dev/null)
+
 #######################
 # For local development
 
@@ -80,19 +84,19 @@ dist: clean browserpass-linux64 browserpass-darwin64 browserpass-openbsd64 brows
 
 .PHONY: configure
 configure:
-	sed -i "s|%%replace%%|$(BIN_DIR)/$(BIN)|" browser-files/chromium-host.json
-	sed -i "s|%%replace%%|$(BIN_DIR)/$(BIN)|" browser-files/firefox-host.json
+	$(SED) -i 's|"path": ".*"|"path": "'"$(BIN_DIR)/$(BIN)"'"|' browser-files/chromium-host.json
+	$(SED) -i 's|"path": ".*"|"path": "'"$(BIN_DIR)/$(BIN)"'"|' browser-files/firefox-host.json
 
 .PHONY: install
 install:
-	install -Dm755 -t "$(BIN_DIR)/" $(BIN)
-	install -Dm644 -t "$(LIB_DIR)/browserpass/" Makefile
-	install -Dm644 -t "$(SHARE_DIR)/licenses/browserpass/" LICENSE
-	install -Dm644 -t "$(SHARE_DIR)/doc/browserpass/" README.md
+	$(INSTALL) -Dm755 -t "$(BIN_DIR)/" $(BIN)
+	$(INSTALL) -Dm644 -t "$(LIB_DIR)/browserpass/" Makefile
+	$(INSTALL) -Dm644 -t "$(SHARE_DIR)/licenses/browserpass/" LICENSE
+	$(INSTALL) -Dm644 -t "$(SHARE_DIR)/doc/browserpass/" README.md
 
-	install -Dm644 browser-files/chromium-host.json   "$(LIB_DIR)/browserpass/hosts/chromium/$(APP_ID).json"
-	install -Dm644 browser-files/chromium-policy.json "$(LIB_DIR)/browserpass/policies/chromium/$(APP_ID).json"
-	install -Dm644 browser-files/firefox-host.json    "$(LIB_DIR)/browserpass/hosts/firefox/$(APP_ID).json"
+	$(INSTALL) -Dm644 browser-files/chromium-host.json   "$(LIB_DIR)/browserpass/hosts/chromium/$(APP_ID).json"
+	$(INSTALL) -Dm644 browser-files/chromium-policy.json "$(LIB_DIR)/browserpass/policies/chromium/$(APP_ID).json"
+	$(INSTALL) -Dm644 browser-files/firefox-host.json    "$(LIB_DIR)/browserpass/hosts/firefox/$(APP_ID).json"
 
 # Browser-specific hosts targets
 

--- a/Makefile
+++ b/Makefile
@@ -187,13 +187,13 @@ hosts-vivaldi-user:
 .PHONY: hosts-brave
 hosts-brave:
 	@case $(OS) in \
-	Linux)      mkdir -p "/etc/opt/brave/native-messaging-hosts/"; \
-	            ln -sfv "$(LIB_DIR)/browserpass/hosts/chromium/$(APP_ID).json" "/etc/opt/brave/native-messaging-hosts/$(APP_ID).json"; \
-	            [ -e "/etc/opt/brave/native-messaging-hosts/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
+	Linux)      mkdir -p "/etc/opt/chrome/native-messaging-hosts/"; \
+	            ln -sfv "$(LIB_DIR)/browserpass/hosts/chromium/$(APP_ID).json" "/etc/opt/chrome/native-messaging-hosts/$(APP_ID).json"; \
+	            [ -e "/etc/opt/chrome/native-messaging-hosts/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
 	            ;; \
-	Darwin)     mkdir -p "/Library/Application Support/BraveSoftware/Brave-Browser/NativeMessagingHosts/"; \
-	            ln -sfv "$(LIB_DIR)/browserpass/hosts/chromium/$(APP_ID).json" "/Library/Application Support/BraveSoftware/Brave-Browser/NativeMessagingHosts/$(APP_ID).json"; \
-	            [ -e "/Library/Application Support/BraveSoftware/Brave-Browser/NativeMessagingHosts/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
+	Darwin)     mkdir -p "/Library/Application Support/Chromium/NativeMessagingHosts/"; \
+	            ln -sfv "$(LIB_DIR)/browserpass/hosts/chromium/$(APP_ID).json" "/Library/Application Support/Chromium/NativeMessagingHosts/$(APP_ID).json"; \
+	            [ -e "/Library/Application Support/Chromium/NativeMessagingHosts/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
 	            ;; \
 	*)          echo "The operating system $(OS) is not supported"; exit 1 ;; \
 	esac
@@ -205,9 +205,9 @@ hosts-brave-user:
 	            ln -sfv "$(LIB_DIR)/browserpass/hosts/chromium/$(APP_ID).json" "${HOME}/.config/BraveSoftware/Brave-Browser/NativeMessagingHosts/$(APP_ID).json"; \
 	            [ -e "${HOME}/.config/BraveSoftware/Brave-Browser/NativeMessagingHosts/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
 	            ;; \
-	Darwin)     mkdir -p "${HOME}/Library/Application Support/BraveSoftware/Brave-Browser/NativeMessagingHosts/"; \
-	            ln -sfv "$(LIB_DIR)/browserpass/hosts/chromium/$(APP_ID).json" "${HOME}/Library/Application Support/BraveSoftware/Brave-Browser/NativeMessagingHosts/$(APP_ID).json"; \
-	            [ -e "${HOME}/Library/Application Support/BraveSoftware/Brave-Browser/NativeMessagingHosts/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
+	Darwin)     mkdir -p "${HOME}/Library/Application Support/Google/Chrome/NativeMessagingHosts/"; \
+	            ln -sfv "$(LIB_DIR)/browserpass/hosts/chromium/$(APP_ID).json" "${HOME}/Library/Application Support/Google/Chrome/NativeMessagingHosts/$(APP_ID).json"; \
+	            [ -e "${HOME}/Library/Application Support/Google/Chrome/NativeMessagingHosts/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
 	            ;; \
 	*)          echo "The operating system $(OS) is not supported"; exit 1 ;; \
 	esac
@@ -329,13 +329,13 @@ policies-vivaldi-user:
 .PHONY: policies-brave
 policies-brave:
 	@case $(OS) in \
-	Linux)      mkdir -p "/etc/opt/brave/policies/managed/"; \
-	            ln -sfv "$(LIB_DIR)/browserpass/policies/chromium/$(APP_ID).json" "/etc/opt/brave/policies/managed/$(APP_ID).json"; \
-	            [ -e "/etc/opt/brave/policies/managed/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
+	Linux)      mkdir -p "/etc/opt/chrome/policies/managed/"; \
+	            ln -sfv "$(LIB_DIR)/browserpass/policies/chromium/$(APP_ID).json" "/etc/opt/chrome/policies/managed/$(APP_ID).json"; \
+	            [ -e "/etc/opt/chrome/policies/managed/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
 	            ;; \
-	Darwin)     mkdir -p "/Library/Application Support/BraveSoftware/Brave-Browser/policies/managed/"; \
-	            ln -sfv "$(LIB_DIR)/browserpass/policies/chromium/$(APP_ID).json" "/Library/Application Support/BraveSoftware/Brave-Browser/policies/managed/$(APP_ID).json"; \
-	            [ -e "/Library/Application Support/BraveSoftware/Brave-Browser/policies/managed/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
+	Darwin)     mkdir -p "/Library/Application Support/Chromium/policies/managed/"; \
+	            ln -sfv "$(LIB_DIR)/browserpass/policies/chromium/$(APP_ID).json" "/Library/Application Support/Chromium/policies/managed/$(APP_ID).json"; \
+	            [ -e "/Library/Application Support/Chromium/policies/managed/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
 	            ;; \
 	*)          echo "The operating system $(OS) is not supported"; exit 1 ;; \
 	esac
@@ -347,9 +347,9 @@ policies-brave-user:
 	            ln -sfv "$(LIB_DIR)/browserpass/policies/chromium/$(APP_ID).json" "${HOME}/.config/BraveSoftware/Brave-Browser/policies/managed/$(APP_ID).json"; \
 	            [ -e "${HOME}/.config/BraveSoftware/Brave-Browser/policies/managed/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
 	            ;; \
-	Darwin)     mkdir -p "${HOME}/Library/Application Support/BraveSoftware/Brave-Browser/policies/managed/"; \
-	            ln -sfv "$(LIB_DIR)/browserpass/policies/chromium/$(APP_ID).json" "${HOME}/Library/Application Support/BraveSoftware/Brave-Browser/policies/managed/$(APP_ID).json"; \
-	            [ -e "${HOME}/Library/Application Support/BraveSoftware/Brave-Browser/policies/managed/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
+	Darwin)     mkdir -p "${HOME}/Library/Application Support/Google/Chrome/policies/managed/"; \
+	            ln -sfv "$(LIB_DIR)/browserpass/policies/chromium/$(APP_ID).json" "${HOME}/Library/Application Support/Google/Chrome/policies/managed/$(APP_ID).json"; \
+	            [ -e "${HOME}/Library/Application Support/Google/Chrome/policies/managed/$(APP_ID).json" ] || echo "Error: the symlink points to a non-existent location" >&2; \
 	            ;; \
 	*)          echo "The operating system $(OS) is not supported"; exit 1 ;; \
 	esac

--- a/README.md
+++ b/README.md
@@ -54,6 +54,8 @@ Primary key fingerprint: EB4F 9E5A 60D3 2232 BB52  150C 12C8 7A28 FEAC 6B20
 
 Unpack the archive. If you decided to compile the application yourself, refer to the [Building the app](#building-the-app) section on how to do so. Once complete, continue with the steps below.
 
+If you are on macOS, first install the necessary tools: `brew install coreutils gnu-sed`.
+
 If you downloaded a release archive with pre-compiled binary, follow these steps to install the app:
 
 ```
@@ -62,14 +64,21 @@ make BIN=browserpass-XXXX configure      # Configure the hosts json files
 sudo make BIN=browserpass-XXXX install   # Install the app
 ```
 
+In addition, both `configure` and `install` targets respect `PREFIX`, `DESTDIR` parameters if you want to customize the install location (e.g. to install to a `$HOME` dir to avoid using `sudo`).
+
+For example, if you are on macOS, you probably want to install Browserpass in `/usr/local/bin`, therefore you need to run:
+
+```
+make BIN=browserpass-darwin64 PREFIX=/usr/local configure      # Configure the hosts json files
+sudo make BIN=browserpass-darwin64 PREFIX=/usr/local install   # Install the app
+```
+
 If you compiled the app yourself, you can omit `BIN` parameter:
 
 ```
 make configure      # Configure the hosts json files
 sudo make install   # Install the app
 ```
-
-In addition, both `configure` and `install` targets respect `PREFIX`, `DESTDIR` parameters if you want to customize the install location (e.g. to install to a `$HOME` dir to avoid using `sudo`).
 
 Finally proceed to the [Configure browsers](#configure-browsers) section.
 
@@ -125,13 +134,17 @@ Remember to check [Hints for configuring gpg](#hints-for-configuring-gpg) on how
 
 The following operating systems provide packages for certain browsers that can be installed using a package manager:
 
--   Arch Linux: [browserpass-chromium](https://www.archlinux.org/packages/community/any/browserpass-chromium/) and [browserpass-firefox](https://www.archlinux.org/packages/community/any/browserpass-firefox/)
+-   Available since Saturday, 13 April: ~Arch Linux: [browserpass-chromium](https://www.archlinux.org/packages/community/any/browserpass-chromium/) and [browserpass-firefox](https://www.archlinux.org/packages/community/any/browserpass-firefox/)~
 
 If you installed a distro package above, you are done!
 
 If something went wrong, or if there's no package for your OS and/or a browser of your choice, proceed with the steps below.
 
-The Makefile (which is also available in `/usr/lib/browserpass/`, if you installed via package manager) contains the following `make` goals to configure the browsers you use:
+First, enter the directory with installed Browserpass, by default it is `/usr/lib/browserpass/`, but if you used `PREFIX` or `DESTDIR` when running `make install`, it might be different for you. For example, on macOS the directory is likely to be `/usr/local/lib/browserpass/`.
+
+See below the list of available `make` goals to configure various browsers.
+
+If you provided `PREFIX` and/or `DESTDIR` while running `make install`, remember that you must provide the same parameters, for example `sudo make PREFIX=/usr/local hosts-chromium`:
 
 | Command                    | Description                                                                |
 | -------------------------- | -------------------------------------------------------------------------- |

